index.js

```diff
@@ -53,42 +59,76 @@ function Release (options, callback) {
     return callback(err)
   }
 
-  // create client
-
-  var client = createClient(options)
-  if (!client) return callback(new Error('missing auth info'))
+  // err if auth info not provided (token or user/pass)
+  if (!getAuth(options)) return callback(new Error('missing auth info'))
 
   // check if commit exists on remote
+  var getCommitOptions = extend(getAuth(options), {
+    method: 'GET',
+    uri: API_ROOT + format('repos/%s/%s/git/commits/%s', options.owner, options.repo, options.target_commitish)
+  })
 
-  var commitOpts = {
-    user: options.owner,
-    repo: options.repo,
-    sha: options.target_commitish
-  }
-
-  client.repos.getCommit(commitOpts, function (err, res) {
+  request(getCommitOptions, function (err, res, body) {
     if (err) {
-      return callback(new Error('Target commitish "' + options.target_commitish +
-        '" not found in ' + options.owner + '/' + options.repo))
+      var errorMessage = format('Target commitish %s not found in %s/%s', options.target_commitish, options.owner, options.repo)
+      return callback(new Error(errorMessage))
     }
 
-    var releaseOpts = prepOptions(options)
+    var releaseOpts = extend(getAuth(options), {
```

Comment:
I actually tried that. Saving it to a variable makes it mutable. This is undesirable because then options collide from request to request.